### ISSUE TITLE
Skeleton2D Node Makes Polygon2D Textures Dark in Android [GLES3]

### ISSUE THREAD

**Issue by samdaiya** [godotengine/godot#27919]
**Godot version:** 3.1.stable.official

**OS/device including version:** Android 8.1 (Realme 1)

**Issue description:** When I create a character using polygons and run it on Android Device everything looks okay but when I assign Skeleton2D Node to it  Textures of Polygon turns Dark.

**Screenshots:**

Output On Linux
![Linux](https://user-images.githubusercontent.com/17422674/55932106-d6ef0c00-5c45-11e9-9fdb-af78e9a6402d.png)

Results On Android
![Android](https://user-images.githubusercontent.com/17422674/55932111-dbb3c000-5c45-11e9-997d-12f8f6828d96.png)

**Steps to reproduce:** Create 2D character with polygons and then Create and Assign Skeleton2D Node to it. Run it on android device.

**Minimal reproduction project:**
[SkeletonBug.zip](https://github.com/godotengine/godot/files/3066971/SkeletonBug.zip)

**Comment by Calinou** [comment]
The reproduction project uses the GLES3 renderer. Can you reproduce this issue if you switch to the GLES2 renderer in the Project Settings? (**Rendering > Quality > Driver > Driver Name**)

**Comment by samdaiya** [comment]
* As you (@Calinou ) suggested I used same Project and Switched to GLES2 Renderer. This time Character With Skeleton2D didn't even showed up. You can see it in following screenshot.


![GLES2](https://user-images.githubusercontent.com/17422674/64271602-36ceb200-cf5b-11e9-9c8b-a6ef8622e799.png)

**Comment by Calinou** [comment]
@samdaiya Skeleton2Ds disappearing in GLES2 seem to be the same issue as #26933.

**Comment by samdaiya** [comment]
What should I do to solve this? Any Workgrounds or permanent Solution?

**Comment by samdaiya** [comment]
> The reproduction project uses the GLES3 renderer. Can you reproduce this issue if you switch to the GLES2 renderer in the Project Settings? (**Rendering > Quality > Driver > Driver Name**)
> 
> Judging by the screenshot, it seems sRGB colors are being forced on the texture with Skeleton2D assigned, but only on Android. This will look correct in 3D, but it will look wrong in 2D.

Should I expect this to be solved in 3.2 or I just distch using New Skeleton2D System?

**Comment by Calinou** [comment]
@samdaiya Since 3.2 is in feature freeze and nobody has been working on this, I don't think this will be solved by the time it's released.

**Comment by samdaiya** [comment]
It's sad to hear. It's really good feature but Only Android have this problem. I've tried to track down problem myself but couldn't. Where should I look to solve this problem. I've checked Polygon2d and Skeleton2D classes but couldn't find any issues in them.

**Comment by Calinou** [comment]
@samdaiya This is likely related to mobile graphics drivers, which are unfortunately difficult to debug (especially if you have no OpenGL experience).

**Comment by samdaiya** [comment]
I'm still learning basic OpenGL. Hopefully it will be solved in 3.2.x. Until then I will use old method for skeleton Animation without Polygon2D.

**Comment by akien-mga** [comment]
Tested in the current master branch (bab91bd943aa5a23bab871e58e9640536cc67b34) on two devices:

- Samsung Galaxy S3 (old, GLES2): the polygon character with Skeleton2D is not displayed at all.
- Xiaomi Pocophone F1 (recent):
  * GLES2: Works fine.
  * GLES3: Same as OP, the character with Skeleton2D appears darker than the other one.

So the bug in OP is GLES3-specific, but old GLES2 devices have an even harder time with Skeleton2D.

**Comment by clayjohn** [comment]
Aiming for the 3.2.1 release

**Comment by samdaiya** [comment]
> Aiming for the 3.2.1 release

Good to know.

**Comment by samdaiya** [comment]
Any update on this? Still facing the problem.

**Comment by Calinou** [comment]
@samdaiya It might be worth checking whether [3.2.4 beta1](https://godotengine.org/article/dev-snapshot-godot-3-2-4-beta-1) solves this issue.

**Comment by lawnjelly** [comment]
Yes in 3.2.4 2d software skinning is supported in GLES2 and GLES3 (and is on by default). This should hopefully solve the problem of skeletons not working on some GLES2 hardware. As to the colors, I'm not sure off hand. The 2d polygons have vertex coloring if I remember correctly. Could be a linear / sRGB discrepancy.

**Comment by samdaiya** [comment]
I downloaded [v3.2.4.beta1.official](https://downloads.tuxfamily.org/godotengine/3.2.4/beta1/) and imported [SkeletonBug](https://github.com/godotengine/godot/files/3066971/SkeletonBug.zip) in it, Now Godot Editor ( GLES2/GLES3 ) is not able to show Polygons2Ds that have Skeleton2D attached to them. Also it is not visible in Android.

**Here's the SS:**
![Polygon2D](https://user-images.githubusercontent.com/17422674/97526609-837f1300-19cf-11eb-8c2b-e4ad5d400985.png)

**Comment by lawnjelly** [comment]
I'll have a look when I get back, there may be another skinning bug to solve (I've only had 3 test projects to test for regressions). But it is unlikely to be a big problem, and if it works on desktop it should work on android (as the shader is quite simple).

Until then you could test these projects on android:
#35772 
https://github.com/godotengine/godot/issues/42576#issuecomment-704429864

**Comment by lawnjelly** [comment]
Ah my mistake, I downloaded skeleton_bug and although I couldn't completely try it here on my old laptop, it doesn't appear to be a skinned mesh, it is just using bones to move objects. In which case the software skinning won't be involved, this may be a completely different bug.

**Comment by samdaiya** [comment]
* I tested both of the projects as you shared and they are working properly. 
* After that i created my own project and synced bones ( I probably forgot to do that in previous project, it was long time ago and I don't remember it ). Seems like it's working properly on both Android & Desktop with both GLES2 & GES3. 
* Thanks @Calinou, @lawnjelly and anyone else who helped to fix it. I can finally start remaking my old game.

* Here's the latest project I created if anyone else want to test it too. -> [SkeletonTest.zip](https://github.com/godotengine/godot/files/5458991/SkeletonTest.zip)
